data-prepper-plugins/rds-source/src/main/java/org/opensearch/dataprepper/plugins/source/rds/schema/ConnectionManagerFactory.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ */
+
+package org.opensearch.dataprepper.plugins.source.rds.schema;
+
+import org.opensearch.dataprepper.plugins.source.rds.RdsSourceConfig;
+import org.opensearch.dataprepper.plugins.source.rds.configuration.EngineType;
+import org.opensearch.dataprepper.plugins.source.rds.model.DbMetadata;
+
+import java.util.List;
+
+public class ConnectionManagerFactory {
+    private final RdsSourceConfig sourceConfig;
+    private final DbMetadata dbMetadata;
+
+    public ConnectionManagerFactory(final RdsSourceConfig sourceConfig, final DbMetadata dbMetadata) {
+        this.sourceConfig = sourceConfig;
+        this.dbMetadata = dbMetadata;
+    }
+
+    public ConnectionManager getConnectionManager() {
+        if (sourceConfig.getEngine() == EngineType.MYSQL) {
```

Comment:
`sourceConfig.getEngine()` is this mandatory field ? We can throw exception is this is not set.

Reply:
Yes, it's MYSQL by default.